src/lib/bit/bit.c

```diff
@@ -332,3 +332,110 @@ bit_count(const uint8_t *data, size_t bit_offset, size_t length)
 
 	return count;
 }
+
+void
+bit_copy_range_reverse(uint8_t *restrict dst, size_t dst_i,
+		       const uint8_t *restrict src, size_t src_i, size_t count)
+{
+	if (count == 0)
+		return;
+
+	assert(src_i >= count - 1);
+	/* Assure the buffers are non-overlapping. */
+	assert(src + (src_i - count + 1) / CHAR_BIT >
+	       dst + (dst_i + count) / CHAR_BIT ||
+	       dst + dst_i / CHAR_BIT > src + src_i / CHAR_BIT);
```

Comment:
Nit: that's not easy to follow. I think we can make it stricter and check `[src ... src_i]` and `[dst ... dst_i + count]` with something similar to what we have in `bit_copy_range`, as we don't use the function with overlapping buffers I see.